util/util_test.go

```diff
@@ -905,3 +907,66 @@ func TestPathSliceSort(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyFs1(t *testing.T) {
+	src := afero.NewMemMapFs()
+	dst := afero.NewMemMapFs()
+
+	fooData := []byte("foo")
+	barData := []byte("bar")
+	bazData := []byte("baz")
+
+	src.MkdirAll("/foo/bar/baz/", 0700)
```

Comment:
*any time* we have something that returns an error, we need to explicitly handle it, or explain why we can't in a comment... there are a bunch of these in here... Sometimes when you see an excessive amount of error handling, it's occasionally a hint that it could be re-written differently. See my main comment at the end.